src/sentry/search/events/fields.py

```diff
@@ -1154,14 +1242,113 @@ def is_accessible(self, acl=None):
         Function(
             "count_miserable",
             required_args=[CountColumn("column"), NumberRange("satisfaction", 0, None)],
-            calculated_args=[{"name": "tolerated", "fn": lambda args: args["satisfaction"] * 4.0}],
+            calculated_args=[
+                {"name": "tolerated", "fn": lambda args, _: args["satisfaction"] * 4.0}
+            ],
             aggregate=[
                 "uniqIf",
                 [ArgValue("column"), ["greater", ["transaction.duration", ArgValue("tolerated")]]],
                 None,
             ],
             default_result_type="number",
         ),
+        Function(
+            "count_miserable_new",
+            required_args=[
+                CountColumn("column"),
+                FunctionAliasArg("threshold_config"),
+            ],
+            aggregate=[
+                "uniqIf",
+                [
+                    ArgValue("column"),
+                    [
+                        "greater",
+                        [
+                            [
+                                "multiIf",
+                                [
+                                    [
+                                        "equals",
+                                        [
+                                            [
+                                                "tupleElement",
+                                                [
+                                                    ArgValue("threshold_config"),
+                                                    1,
+                                                ],
+                                            ],
+                                            "'lcp'",
+                                        ],
+                                    ],
+                                    "measurements[lcp]",
+                                    # [
+                                    #     "equals",
+                                    #     [
+                                    #         [
+                                    #             "tupleElement",
+                                    #             [
+                                    #                 ArgValue("threshold_config"),
+                                    #                 1,
+                                    #             ],
+                                    #         ],
+                                    #         "'fcp'",
+                                    #     ],
+                                    # ],
+                                    # "measurements[fcp]",
```

Comment:
I'm not able to get multiIf working with multiple statements- it just results in a connection timeout in snuba `Max retries exceeded with url: /query`, need to investigate this further. But in the meantime maybe I should just move forward with support for lcp and duration

Reply:
As mentioned offline, maybe it's worth looking into using `transform` instead.